### PR TITLE
[wasm] `JSImport` generator `AllowUnsafeBlocks` diagnostic

### LINKED ISSUE #72760: [wasm] implement `JSImport` generator `AllowUnsafeBlocks` diagnostic
Same as https://github.com/dotnet/runtime/pull/72650

### PR DESCRIPTION
**Pull request by pavelsavara**
- added `JSImportAttribute requires unsafe code` + unit test
- fixed Diagnostic IDs + reservation
- moved `System.Runtime.InteropServices.JavaScript.UnitTests` to subfolder

Fixes https://github.com/dotnet/runtime/issues/72760

### CODE REVIEW

File: src/libraries/System.Runtime.InteropServices.JavaScript/gen/JSImportGenerator/JSExportGenerator.cs [hunk 1]
```diff
@@ -79,16 +81,33 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
...
                 .Select(static (options, ct) => new JSGeneratorOptions(options.GlobalOptions));
 
+            IncrementalValueProvider<StubEnvironment> stubEnvironment = context.CreateStubEnvironmentProvider();
+
+            // Validate environment that is being used to generate stubs.
+            context.RegisterDiagnostics(stubEnvironment.Combine(attributedMethods.Collect()).SelectMany((data, ct) =>
+            {
+                if (data.Right.IsEmpty // no attributed methods
+                    || data.Left.Compilation.Options is CSharpCompilationOptions { AllowUnsafe: true } // Unsafe code enabled
+                    || data.Left.TargetFramework != TargetFramework.Net) // Non-.NET 5 scenarios use forwarders and don't need unsafe code
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Runtime.InteropServices.JavaScript/gen/JSImportGenerator/JSExportGenerator.cs:101]: > || data.Left.TargetFramework != TargetFramework.Net) // Non-.NET 5 scenarios use forwarders and don't need unsafe code

Is this also true for the JS source generator? I assumed it doesn't use forwarders at all.

Comment (AaronRobinsonMSFT) [reply src/libraries/System.Runtime.InteropServices.JavaScript/gen/JSImportGenerator/JSExportGenerator.cs:101]: Same for JSImport.

**Review by AaronRobinsonMSFT — COMMENTED** [review]
/cc @jkoritzinsky and @elinor-fung

File: src/libraries/System.Runtime.InteropServices.JavaScript/gen/JSImportGenerator/JSExportGenerator.cs [hunk 2]
```diff
@@ -53,6 +53,7 @@ public static class StepNames
 
         public void Initialize(IncrementalGeneratorInitializationContext context)
         {
+            // Collect all methods adorned with LibraryImportAttribute
```
Comment (radekdoulik) [inline src/libraries/System.Runtime.InteropServices.JavaScript/gen/JSImportGenerator/JSExportGenerator.cs:56]: typo? should say JSExportAttribute instead?

**Review by radekdoulik — APPROVED** [review]
Besides the possible typo it LGTM

**Pull request merged**